exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+import org.apache.drill.exec.expr.TypeHelper;
+import org.apache.drill.exec.expr.holders.BigIntHolder;
+import org.apache.drill.exec.expr.holders.NullableFloat8Holder;
+import org.apache.drill.exec.expr.holders.ObjectHolder;
+import org.apache.drill.exec.expr.holders.ValueHolder;
+import org.apache.drill.exec.ops.FragmentContext;
+import org.apache.drill.exec.physical.config.StatisticsMerge;
+import org.apache.drill.exec.record.AbstractSingleRecordBatch;
+import org.apache.drill.exec.record.MaterializedField;
+import org.apache.drill.exec.record.RecordBatch;
+import org.apache.drill.exec.record.TransferPair;
+import org.apache.drill.exec.record.VectorContainer;
+import org.apache.drill.exec.record.VectorWrapper;
+import org.apache.drill.exec.vector.ValueVector;
+import org.apache.drill.exec.vector.DateVector;
+import org.apache.drill.exec.vector.BigIntVector;
+import org.apache.drill.exec.vector.NullableBigIntVector;
+import org.apache.drill.exec.vector.NullableFloat8Vector;
+import org.apache.drill.exec.vector.NullableVarBinaryVector;
+import org.apache.drill.exec.vector.VarCharVector;
+import org.apache.drill.exec.vector.complex.MapVector;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+import com.clearspring.analytics.stream.cardinality.HyperLogLog;
+
+public class StatisticsMergeBatch extends AbstractSingleRecordBatch<StatisticsMerge> {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(StatisticsMergeBatch.class);
+  private Map<String, String> functions;
+  private boolean first = true;
+  private boolean finished = false;
+  private int schema = 0;
+  private int recordCount = 0;
+  private List<String> keyList = null;
+  private Map<MaterializedField, ValueVector> dataSrcVecMap = null;
+  // Map of non-map fields to VV in the incoming schema
+  private Map<MaterializedField, ValueVector> copySrcVecMap = null;
+  private Map<String, Map<String, ValueHolder>> aggregationMap = null;
+  public StatisticsMergeBatch(StatisticsMerge popConfig, RecordBatch incoming,
+                              FragmentContext context) throws OutOfMemoryException {
+    super(popConfig, context, incoming);
+    this.functions = new HashMap<>();
+    this.aggregationMap = new HashMap<>();
+
+    /*for (String key : popConfig.getFunctions()) {
+      aggregationMap.put(key, new HashMap<String, ValueHolder>());
+      if (key.equalsIgnoreCase("statcount") || key.equalsIgnoreCase("nonnullstatcount")) {
+        functions.put(key, "sum");
+      } else if (key.equalsIgnoreCase("hll")) {
+        functions.put(key, "hll_merge");
+      } else if (key.equalsIgnoreCase("sum_width")) {
+        functions.put(key, "avg_width");
+      }
+    }*/
+    for (String key : popConfig.getFunctions()) {
```

Comment:
Yes. The refactored code should allow us to do so. For stats agg, we use the passed in functions (in AnalyzePRule). We do the same now for stats merge instead of hard-coded functions inside the StatisticsMergeBatch.